app/decorators/controllers/badges_engine/application_controller_decorator.rb

```diff
@@ -4,12 +4,12 @@
   private
 
   def restrict_access
-    redirect_to session[:last_moudle] unless
+    redirect_to session[:last_module] unless
```

Comment:
in the other controller you changed this to current_user, why the difference here, also, seeing this, I would recommend you to create a concern.... module.... service maybe.... or just a helper method for this two methods as you're currently duplicating logic